src/Plugin/Field/FieldType/ProjectGenusTypeItem.php

```diff
@@ -0,0 +1,270 @@
+        'table_alias_mapping' => ['scinameprop' => 'projectprop'],
+        'delete_if_empty' => TRUE,
+      ]),
+      new ChadoIntStoragePropertyType($entity_type_id, self::$id, 'sciname_rank', 'OBCS:0000117', [
+        'action' => 'store',
+        'path' => 'project.project_id>scinameprop.project_id;rank',
+        'table_alias_mapping' => ['scinameprop' => 'projectprop'],
+      ]),
+      new ChadoIntStoragePropertyType($entity_type_id, self::$id, 'sciname_type_id', 'schema:additionalType', [
+        'action' => 'store',
+        'path' => 'project.project_id>scinameprop.project_id;type_id',
+        'table_alias_mapping' => ['scinameprop' => 'projectprop'],
+      ]),
+    ];
+  }
+
+  /**
+   * We need to set the type_id property value to match the cvterm_id.
+   *
+   * To do this we'll override the tripalValuesTemplate() and give the
+   * `type_id` property a default value.
+   *
+   * {@inheritDoc}
+   *
+   * @see \Drupal\tripal\TripalField\TripalFieldItemBase::tripalValuesTemplate()
+   */
+  public function tripalValuesTemplate($field_definition, $default_value = NULL) {
+    $idSpace_manager = \Drupal::service('tripal.collection_plugin_manager.idspace');
+
+    // Use the parent method to get a template values array.
+    $prop_values = parent::tripalValuesTemplate($field_definition, $default_value);
+
+    // Term: genus.
+    $idSpace = $idSpace_manager->loadCollection('TAXRANK');
+    $genus_term = $idSpace->getTerm('0000005');
+    // Term: scientific name.
+    $idSpace = $idSpace_manager->loadCollection('NCBITaxon');
+    $sciename_term = $idSpace->getTerm('scientific_name');
+
+    // FIX the type_id for both our properties using the terms above.
+    foreach ($prop_values as $index => $prop_value) {
+      if ($prop_value->getKey() == 'genus_type_id') {
+        $prop_values[$index]->setValue($genus_term->getInternalId());
+      }
+      elseif ($prop_value->getKey() == 'sciname_type_id') {
+        $prop_values[$index]->setValue($sciename_term->getInternalId());
+      }
+    }
+
+    return $prop_values;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getConstraints() {
+    $constraints = parent::getConstraints();
+
+    /**
+     * Ensure that the value entered is not larger then the max length.
+     * @code
+     * if ($max_length = $this->getSetting('max_length')) {
+     *   $constraint_manager = \Drupal::typedDataManager()->getValidationConstraintManager();
+     *   $constraints[] = $constraint_manager->create('ComplexData', [
+     *     'value' => [
+     *       'Length' => [
+     *         'max' => $max_length,
+     *         'maxMessage' => t('%name: may not be longer than @max characters.', [
+     *           '%name' => $this
+     *           ->getFieldDefinition()
+     *           ->getLabel(),
+     *           '@max' => $max_length,
+     *         ]),
+     *       ],
+     *     ],
+     *   ]);
+     * }
+     * @endcode
+     */
```

Comment:
I believe this applies to text fields, and since the field is now a select box, it's safe to remove it.